src/server.js

```diff
@@ -76,6 +76,48 @@ export class Server {
 
   /**
    * Submits a transaction to the network.
+   *
+   * Warning: sometimes a transaction will "succeed" (without throwing an error)
+   * but the transaction won't appear on the network.
+   *
+   * For example, a user submits a `manageOffer` transaction that would
+   * result in a user receiving less than 0.0000001 of a token. The transaction
+   * won't be logged on the network, yet `submitTransaction` will resolve to this:
```

Comment:
The wording is very confusing in 1st paragraph and 2nd paragraph is not true. The example of `manageOffer` above is actually processed by the network and is included in the ledger. Also (as @orbitlens noted) you should check deserialize into `TransactionResult`.